pftools/python/parflow/tools/builders.py

```diff
@@ -705,47 +728,50 @@ def print_as_table(self, props_in_header=True, column_separator='  '):
 class SubsurfacePropertiesBuilder(TableToProperties):
 
     def __init__(self, run=None):
-        if run is not None:
-            self.run = run
-        self.output = {}
-        self.name_registration = {}
-        self.column_index = {}
-        self.props_in_row_header = True
-        self.table_comments = []
-        yaml_key_def = os.path.join(
-            os.path.dirname(__file__), 'ref/table_keys.yaml')
-        with open(yaml_key_def, 'r') as file:
-            self.definition = yaml.safe_load(file)
+        self.reference_file = 'ref/table_keys.yaml'
```

Comment:
It might be a little cleaner if this is a property too (then you don't need to define it before calling `super()`.